mariamihaelafainarea/assignment4/Assignment4

```diff
@@ -0,0 +1,78 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment4;
+
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ *
+ * @author maria
+ */
+public class Assignment4 {
+
+    int n,m;
+    boolean[][] matrix;
+    DisjointSet d;
+    //false - water
+    //true - land
+    //Please return number of islands ! 
+
+    public Assignment4(int n, int m, boolean[][] matrix) {
+        this.n = n;
+        this.m = m;
+        this.matrix = matrix;
+    }
+    //First, assign for each cell a different set
+    public void putIntoDifferentSets() {
+        d = new DisjointSet();
+        for (int i = 0; i < n*m; i++) {
+            d.makeSet(i);
+        }
+    }
+    //Then, the flood fill alg will get together
+    //the cells in the neighbourhood => set of islands
+    public void applyFloodFillOnMe(boolean[][] verified) {
+        FloodFill f = new FloodFill();
+        for (int i = 0; i < n ;i++) {
+            for(int j = 0; j < m; j++) {
+                f.floodFill(d, verified, matrix, n, m,i,j);
+            }
+        }
+    
+    }
+    //to obtain the number of islands
+    //we get the parent nodes of all sets 
+    //that were processed.
+    //if the node isn't verrified there
+    //is water, not land
+    public int getNrOfDifferentSetsWithIslands(boolean[][] verified) {
```

Comment:
This can be optimised: you don't have to yet again iterate through the whole matrix. You can use the DisjointSet to get immediate answer how many sets it holds.

Reply:
Done it ! i have solved it by iterating through the whole map of DisjointSet